server/server.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"log"
+	"net"
+	"net/rpc"
+	"sync"
+
+	"github.com/lukedmor/conduit_tiny/common"
+)
+
+type Provider struct {
+	cs *conduitServer
+}
+
+func (p *Provider) Join(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	p.cs.providers[a.Addr] = true
+	return nil
+}
+
+func (p *Provider) Leave(a *common.ProviderJoinLeaveArgs, reply *common.Nothing) error {
+	delete(p.cs.providers, a.Addr)
+	return nil
+}
+
+type Client struct {
+	cs *conduitServer
+}
+
```

Comment:
You could ping the server but what would you do if the server is down? The server is in charge of keeping track of providers and clients. It seems like it could be a major point of failure.